src/coreclr/tools/aot/ILCompiler.RyuJit/JitInterface/CorInfoImpl.RyuJit.cs

```diff
@@ -2404,6 +2404,9 @@ private bool getObjectContent(CORINFO_OBJECT_STRUCT_* objPtr, byte* buffer, int
         private CORINFO_OBJECT_STRUCT_* getRuntimeTypePointer(CORINFO_CLASS_STRUCT_* cls)
         {
             TypeDesc type = HandleToObject(cls);
+            if (type.IsCanonicalSubtype(CanonicalFormKind.Any))
```

Comment:
Should this be an assert?

If the JIT asks for this, it sounds like a bug somewhere else.

Reply:
> If the JIT asks for this, it sounds like a bug somewhere else.

Isn't just a matter where to check it? like we discussed in https://github.com/dotnet/runtime/issues/101175#issuecomment-2060983125
it was implemented in the first place on VM side and that's why JIT doesn't check it on all uses of getRuntimeTypePointer on its side

Reply:
Otherwise all uses needs to be guarded with getClassAttrs + potential bugs if a future use forgets it + extra SPMI query